cpp/oneapi/dal/backend/primitives/test/ndarray.cpp

```diff
@@ -489,6 +507,41 @@ TEST_M(ndarray_test, "can assign ndarray", "[ndarray]") {
     sycl::free(x_ptr, queue);
 }
 
+TEST_M(ndarray_test, "can be flattened with host usm", "[ndarray]") {
+    DECLARE_TEST_POLICY(policy);
+    auto& queue = policy.get_queue();
+    auto [x, event] = ndarray<float, 2>::ones(queue, { 7, 5 }, sycl::usm::alloc::host);
+    event.wait_and_throw();
+
+    auto raw_arr = x.flatten();
+
+    for (std::int64_t i = 0; i < x.get_count(); ++i) {
+        REQUIRE(raw_arr[i] == 1.0f);
+    }
+}
+
+TEST_M(ndarray_test, "can be flattened with device usm", "[ndarray]") {
+    DECLARE_TEST_POLICY(policy);
+    auto& queue = policy.get_queue();
+    constexpr std::int64_t m = 7;
+    constexpr std::int64_t n = 5;
+    INFO("CHECK");
+    auto [x, event] = ndarray<float, 2>::ones(queue, { m, n }, sycl::usm::alloc::device);
+    event.wait_and_throw();
+
+    const auto raw_arr = x.flatten(queue);
+
+    REQUIRE(raw_arr.get_count() == x.get_count());
+    INFO("CHECK");
```

Comment:
This INFO is useless, remove it or figure out more descriptive message

Reply:
Fixed